src/helpers/tee.ts

```diff
@@ -506,15 +628,15 @@ export const teeKeySharesRetrieve = async (
         return res.status === TEE_RETRIEVE_STATUS_SUCCESS && res.keyshare_data
           ? res.keyshare_data.split("_")[1]
           : undefined
-      } catch {
-        errors.push("Enclave not available")
+      } catch (err: any) {
+        errors.push(JSON.stringify(err))
       }
     }),
   )
 
   shares = shares.filter((x) => x !== undefined)
   if (shares.length < SSSA_THRESHOLD) {
-    throw new Error(`${Errors.TEE_RETRIEVE_ERROR} - Shares could not be retrieved: ${errors[0]}`)
+    throw { error: `${Errors.TEE_RETRIEVE_ERROR} - Shares could not be retrieved`, details: errors }
```

Comment:
I'm not sure anyone uses this fn outside of Ternoa's internal use, but this exception object replacing the string error, might need to be marked as breaking changes in case.

Reply:
reverted to be string as before